src/pages/signup-page/SignUp.js

```diff
@@ -1,31 +1,91 @@
 import "../../styles/login&signup.css";
+import {Link} from 'react-router-dom'
+import axios from "axios";
+import { useState } from "react";
+import {useNavigate} from "react-router-dom"
 import {Navigation} from "../../components/navigation/navigation";
+import { useAuthContext } from "../../contexts/auth-context";
+
+
 export default function SignUp() {
+  const navigate = useNavigate()
+  const {authCred, setAuthCred} = useAuthContext()
+  const [signupCred, setSignupCred] = useState({
+    firstname : "",
+    lastname : "",
+    email : "",
+    password : "",
+    confirmPassword : ""
+  })
+
+  const signupCredHandler = async({firstname, lastname, email, password}) => {
+    try{
+      const {data : {encodedToken}} = await axios.post("/api/auth/signup", {
+        firstname : firstname,
+        lastname : lastname,
+        email : email,
+        password : password
+      })
+      localStorage.setItem("TOKEN", encodedToken)
+      setAuthCred({...authCred, 
+        authToken : encodedToken,
+        authStatus : true
+      })
+      navigate("/")
+    } catch(err){
+      console.log(err)
+    }
+  }
+
   return (
     <div>
       <Navigation />
+      <form onSubmit = {(e)=>{
+        e.preventDefault()
+        signupCredHandler(signupCred)
+      }}>
       <div className="credentials signup__form">
         <h1>Signup</h1>
         <div className="cred__input-field">
           <label>First Name</label>
-          <input placeholder="first name" type="text"></input>
+          <input 
+          placeholder="first name" 
+          type="text"
+          onChange={(e) => setSignupCred({...signupCred, firstname : e.target.value})}
+          ></input>
         </div>
         <div className="cred__input-field">
           <label>Last Name</label>
-          <input placeholder="last name" type="text"></input>
+          <input 
+          placeholder="last name" 
+          type="text"
+          onChange={(e) => setSignupCred({...signupCred, lastname : e.target.value})}
+          ></input>
         </div>
         <div className="cred__input-field">
           <label>Email adddress</label>
-          <input placeholder="akash@neog.com" type="email"></input>
+          <input 
+          placeholder="akash@neog.com" 
+          type="email"
+          onChange={(e) => setSignupCred({...signupCred, email : e.target.value})}
+          ></input>
         </div>
         <div className="cred__input-field password__field">
           <label>Password</label>
-          <input placeholder="password" type="password"></input>
+          <input 
+          placeholder="password" 
+          type="password"
+          onChange={(e) => setSignupCred({...signupCred, password : e.target.value})}
+          ></input>
           <i className="bx bx-show"></i>
```

Comment:
When time permits you can implement the show password functionality here, won't take much time